distributed/shuffle/_shuffle.py

```diff
@@ -116,7 +116,7 @@ def rearrange_by_column_p2p(
             f"p2p requires all column names to be str, found: {unsupported}",
         )
 
-    name = f"shuffle-p2p-{token}"
+    name = f"shuffle_p2p-{token}"
```

Comment:
This is an interesting change I noticed. The previous `shuffle-p2p` would be improperly split such that the prefix is `shuffle` instead of `shuffle-p2p` (due to the `-`)